technologydata/sources.py

```diff
@@ -208,6 +222,378 @@ def process(self, trusted_execution: bool = False) -> None:
+                    self.details.loc[index, "url_archived"] = archived_url
+
+        # Update the existing .csv file with the new attributes
+        if self.path is not None:
+            output_path = pathlib.Path(self.path, output_file_name)
+        else:
+            return None
+
+        # Drop the first column source_name
+        details_df = self.details.copy()
+        details_df = details_df.drop(details_df.columns[0], axis=1)
+
+        # Export to sources.csv
+        details_df.to_csv(
+            output_path, quoting=csv.QUOTE_ALL, quotechar='"', index=False
+        )
+
+        logger.info(f"Updated details saved to {output_path}.")
+
+    @staticmethod
+    def store_snapshot_on_wayback(
+        url_to_archive: str,
+    ) -> tuple[Any, bool | None, str | None] | None:
+        """
+        Store a snapshot of the given URL on the Wayback Machine and extract the timestamp.
+        This method captures the specified URL using the Wayback Machine and retrieves the
+        corresponding archive URL along with a formatted timestamp. The timestamp is extracted
+        from the archive URL and converted to a more readable format.
+
+        Parameters
+        ----------
+        url_to_archive : str
+            The URL that you want to archive on the Wayback Machine.
+
+        Returns
+        -------
+        tuple[str, bool, str] | None
+            A tuple containing the archive URL, a boolean indicating if a new capture was conducted (if the boolean is True,
+            archive.org conducted a new capture. If it is False, archive.org has returned a recently cached capture
+            instead, likely taken in the previous minutes) and the formatted timestamp (with format YYYY-MM-DD hh:mm:ss)
+            if the operation is successful. Returns None if the timestamp cannot be extracted due to a ValueError (e.g.,
+            if the expected substrings are not found in the archive URL).
+
+        """
+        archive_url = savepagenow.capture_or_cache(url_to_archive)
+        try:
+            # The timestamp is between "web/" and the next "/" afterward
+            # Find the starting index of "web/"
+            start_index = archive_url[0].index("web/") + len("web/")
+            # Find the ending index of the timestamp by locating the next "/" after the start_index
+            end_index = archive_url[0].index("/", start_index)
+            # Extract the timestamp substring
+            timestamp = archive_url[0][start_index:end_index]
+            output_timestamp = td.Utils.change_datetime_format(
+                timestamp,
+                td.DateFormatEnum.SOURCES_CSV,
+            )
+            return archive_url[0], archive_url[1], output_timestamp
+        except ValueError:
+            # If "web/" or next "/" not found, return empty string
+            return None
+
+    def download_file_from_wayback(self) -> list[pathlib.Path | None]:
```

Comment:
```suggestion
    def download_files_from_wayback(self, download_directory: str | pathlib.Path) -> list[pathlib.Path | None]:
```

Reply:
Instead of relying on `self.path`, ask the user to specify a download directory explicitly and download it to there.